### DISCUSSION TITLE
Installed on fresh PC - VS2022 and Wix v3 and have build problems

### DISCUSSION THREAD

**Discussion by CSHinUK** [wixtoolset/issues#7900] [Questions]
I have a lot of existing Wix installers and recently I got a new development PC.  Only VS 2022 is installed and Wix v3 extension.  However I can't build any of the existing installers that I have.  I get this error each time and I'm not sure what I'm missing.

Severity	Code	Description	Project	File	Line	Suppression State
Error		The WiX Toolset v3.11 build tools must be installed to build this project. Could not find wix.targets at 'C:\Program Files (x86)\MSBuild\Microsoft\WiX\v3.x\'. To download WiX Toolset v3.11, see https://wixtoolset.org/releases/v3.11/stable	XXXXXXXXXServiceInstaller	C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\WiX\v3.x\Wix.targets	10

**Comment by wyrdfish** [comment]
It looks like a pretty good error message to me, go to https://wixtoolset.org/releases/v3.11/stable and install the toolset from the .exe. You need the tools as well as the VS extension.

https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311.exe

**Reply by CSHinUK** [reply]
Yep that's obvious but I should have expanded.  Tried that, takes you to a link for a 2019 download which requires .NET 3.5.1.  I've installed the latest VS extension that supports 2022 so I assumed that was an old error and thought I'd check on here before downloading an old version of .NET.

**Comment by robmen** [comment]
The last WiX v3 build is over four years old (2019), and that was just a security release. The last v3 release with any feature development was over 6.5 years ago (2017).

[Unless you are on a support contract](https://www.firegiant.com/products/fg-wix/), you should be looking at upgrading to WiX v4.

**Comment by CSHinUK** [comment]
Thanks, it's now working.  Had to raise a request to have .NET 3.5.1 installed by our IT dept..  A bit misleading when trying to get setup with an extension that has been updated for VS2022 but still has dependency for an out of support version of .NET

**Reply by barnson** [reply]
.NET 3.5 SP1 is supported for another five years.

**Comment by chrpai** [comment]
How is it that you have "a lot" of WiX installers and yet you don't know that you need WiX and Votive and that WiX has a dependency on .NET 3.5?  I've written "a lot" (hundreds) of WiX installers over the past 15 years and have had to setup my development and build environments countless times during that period so it's practically impossible to not know what the stack looks like.

As Rob said, if you want to be able to install VS2022, heatwave and nothing else then WiX v4 is the way to go.   WiX isn't "installed" anymore it's just a nuget reference and it doesn't have a dependency on .NET 3.5 anymore.  This is 12 month old information so I'd understand a lot that this is news to people.
